aten/src/ATen/native/vulkan/VulkanAten.cpp

```diff
@@ -0,0 +1,297 @@
+
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/OpaqueTensorImpl.h>
+#include <ATen/native/UpSample.h>
+#include <ATen/native/utils/ParamUtils.h>
+#include <ATen/native/vulkan/VulkanDebugUtils.h>
+
+#ifdef USE_VULKANGL
+#include <ATen/native/vulkan/VulkanGL.h>
+#endif
+
+#ifdef USE_VULKAN
+#include <ATen/native/vulkan/VulkanVulkan.h>
+#endif
+
+namespace at {
+namespace native {
+
+template <typename T>
+struct CAFFE2_API IntrusivePtrTargetWrapper : c10::intrusive_ptr_target {
+ private:
+  T target_;
+
+ public:
+  IntrusivePtrTargetWrapper() = delete;
+  IntrusivePtrTargetWrapper(const T& target) : target_(target) {}
+  IntrusivePtrTargetWrapper(T&& target) : target_(std::move(target)) {}
+
+  T& get_target() {
+    return target_;
+  }
+};
+
+#ifdef USE_VULKAN
+using VTensor = at::native::vulkan::details::vulkan::VulkanVulkanTensor;
+#endif
+#ifdef USE_VULKANGL
+using VTensor = at::native::vulkan::details::gl::VulkanGLTensor;
+#endif
```

Comment:
I will add mutually exclusive logic with priority to USE_VULKAN.
I am thinking if to keep GLES option, should it be a separate Backend or it can be switch inside USE_VULKAN as a fallback?